Prevent a shop from trading tools deceptively.

jhcole opened the pull request:
This limits the tools a shop will accept or give to those whose wear is less
than or equal to the tool shown in the Owner needs or Owner gives lists. This
protects both owners and customers from unfair trades. Owners are protected
from receiving worn out tools when a new tool is placed in the needs list.
Customers are protected from receiving worn out tools when a new one is placed
in the give list.

fozolo committed:
Prevent a shop from trading tools deceptively.

This limits the tools a shop will accept or give to those whose wear is less
than or equal to the tool shown in the Owner needs or Owner gives lists. This
protects both owners and customers from unfair trades. Owners are protected
from receiving worn out tools when a new tool is placed in the needs list.
Customers are protected from receiving worn out tools when a new one is placed
in the give list.

SmallJoker commented on shop.lua:
```diff
@@ -5,6 +5,30 @@
+		if list_stack:get_name()  == stack:get_name()  and
+		   list_stack:get_count() >= stack:get_count() and
+		   list_stack:get_wear()  <= stack:get_wear() then
+			return true, i
```
No need for two return values. Once a stack is found, an `if` check would accept it because it is non-nil and non-false.
```lua
if 5 then
	print("foobar")
end

--> Outputs "foobar"
```
Otherwise excellent formatted code!

EDIT: Also store the stack name in a local variable to speed up the execution time. This function is quite often called for each item field in "Owner gives" and "Owner wants"

jhcole replied:
These are both good catches. I've made the requested changes. ATM they are in a separate commit for clarity but I will squash them into one upon request.

fozolo committed:
Make suggested changes.

  - Simplifies return values by returning only a single value for success and
    nothing for failure.

  - Stores the inventory list in a local to prevent redundant retrievals.

SmallJoker commented:
Successfully tested: works as expected.

Merged as 6f9cc00
